projects/laji/src/app/shared-modules/select/select-subcategories/select-subcategories.component.ts

```diff
@@ -0,0 +1,525 @@
+import {
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  ElementRef,
+  EventEmitter,
+  Input,
+  OnChanges,
+  OnDestroy,
+  OnInit,
+  Output,
+  ViewChild
+} from '@angular/core';
+import { interval as ObservableInterval, Subject } from 'rxjs';
+import { debounceTime, take, takeUntil } from 'rxjs/operators';
+import { FilterService } from '../../../shared/service/filter.service';
+import { WarehouseQueryInterface } from '../../../shared/model/WarehouseQueryInterface';
+import { SelectOptions as SelectComponentOptions } from '../select/select.component';
+
+export interface SelectOptions extends SelectComponentOptions {
+  category: string;
+}
+
+export interface SelectedOptions {
+  [category: string]: Array<string|SelectOptions>;
+}
+
+@Component({
+  selector: 'laji-select-subcategories',
+  templateUrl: './select-subcategories.component.html',
+  styleUrls: ['./select-subcategories.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class SelectSubcategoriesComponent implements OnInit, OnChanges, OnDestroy {
+  private unsubscribe$ = new Subject<null>();
+
+  @Input() query: WarehouseQueryInterface;
+  @Input() options: [{[key: string]: SelectOptions[]}];
+  @Input() title: string;
+  @Input() filterPlaceHolder = 'Search...';
+  @Input() useFilter = true;
+  @Input() selected: SelectedOptions = {};
```

Comment:
There seems to be inconsistencies in the way this property is used. It's good idea to type this and the output (selectedChange) so that we can leverage the type checking